waiter/src/waiter/main.clj

```diff
@@ -117,6 +118,7 @@
     (reify Thread$UncaughtExceptionHandler
       (uncaughtException [_ thread throwable]
         (log/error throwable (str (.getName thread) " threw exception: " (.getMessage throwable))))))
+  (jvm-metrics/instrument-jvm)
```

Comment:
`instrument-jvm` creates gauges.  The overhead comes when the gauges are queried, at which point each gauge calls its backing function to return the current value.